internal/visitors/visitor_func.go

```diff
@@ -42,18 +41,32 @@ func (v funcVisitor) Visit(n ast.Node) ast.Visitor {
 	case *ast.FuncType:
 		// Should not need to declare any non-local definitions in the type params
 		// if node.TypeParams != nil {
-		// 	Walk(v, node.TypeParams)
-		// }
-
-		// Should not need to declare any non-local definitions in the params
-		// if node.Params != nil {
-		// 	Walk(v, node.Params)
+		// 	ast.Walk(v, node.TypeParams)
 		// }
 
 		// Types can create new interfaces and/or types,
 		// so we need to visit them and potentially declare new non-local symbols
 		if node.Results != nil {
-			ast.Walk(v, node.Results)
+			// TODO: for now skip named return idents as theyre not modeled in SCIP,
+			// but we also don't have handeling for attaching hovers to locals in scip-go yet.
```

Comment:
To me, this seems like the problem which needs fixing first. https://github.com/sourcegraph/scip/blob/main/scip.proto#L187-L197 -- After all, SCIP allows attaching documentation to locals just fine, so the parameter should be a local with documentation rather than a global (assuming my point about it not being referenceable outside the file of the definition being correct)

Reply:
Agreed that should be fixed, the current approach is the tunnel i ended up going down. Still seems reasonable to me though given parameters are explicitly defined as non-local in SCIP

Reply:
Languages that support named arguments (like Scala) have global parameter symbols.


```scala
case class User(name: String, age: Int)

// Some other file, named arguments flip order
User(age = 42, name = "John")
```

I agree with @varungandhi-src that we should use local symbols if Go parameters can't be referenced outside of the file. In general, global symbols are much more expensive to index than local symbols.

Reply:
For future readers, here's the canonical link for the above link, since main has since drifted: https://github.com/sourcegraph/scip/blob/e02b917e1f3ff6e958c7df78e06d84d637098862/scip.proto#L187-L197